source/module_esolver/esolver_ks_pw.h

```diff
@@ -56,23 +56,14 @@ class ESolver_KS_PW : public ESolver_KS<T, Device>
     // Init Global class
     void Init_GlobalC(Input& inp, UnitCell& ucell, pseudopot_cell_vnl& ppcell);
 
-    /// @brief allocate psi_init the new psi_initializer
-    void allocate_psi_init();
-
-    /// @brief initialize psi
-    void initialize_psi();
-
   protected:
     //! hide the psi in ESolver_KS for tmp use
     psi::Psi<std::complex<double>, base_device::DEVICE_CPU>* psi = nullptr;
 
-  private:
-    // psi_initializer<T, Device>* psi_init = nullptr;
-    // change to use smart pointer to manage the memory, and avoid memory leak
-    // while the std::make_unique() is not supported till C++14,
-    // so use the new and std::unique_ptr to manage the memory, but this makes new-delete not symmetric
-    std::unique_ptr<psi_initializer<T, Device>> psi_init;
+    // psi_initializer controller
+    psi::InitController<T, Device>* p_initcontroller = nullptr;
```

Comment:
how about change p_initcontroller to p_init_psi or other names that are shorter?